clients/rospy/src/rospy/core.py

```diff
@@ -145,7 +145,25 @@ def rospywarn(msg, *args):
     _rospy_logger.warn(msg, *args)
 
 
-def _base_logger(msg, name=None, throttle=None, level='info'):
+def _base_logger(msg, *args, **kwargs):
+
+    if '_logger_name' in kwargs:

```

Comment:
Could refactor:
`
for (name,param) in [('_logger_name', name),('throttle', throttle), ('_logger_level', level)]:
  try:
    param = kwargs.pop(name)
  except KeyError:
   param = None
`